recipes/corrade/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+def sort_libs(correct_order, libs, lib_suffix='', reverse_result=False):
+    # Add suffix for correct string matching
+    correct_order[:] = [s.__add__(lib_suffix) for s in correct_order]
+
+    result = []
+    for expectedLib in correct_order:
+        for lib in libs:
+            if expectedLib == lib:
+                result.append(lib)
+
+    if reverse_result:
+        # Linking happens in reversed order
+        result.reverse()
+
+    return result
+
+
+class CorradeConan(ConanFile):
+    name = "corrade"
+    description = "Corrade is a multiplatform utility library written in C++11/C++14."
+    topics = ("conan", "corrade", "magnum", "filesystem", "console", "environment", "os")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://magnum.graphics/corrade"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    short_paths = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_deprecated": [True, False],
+        "with_interconnect": [True, False],
+        "with_main": [True, False],
+        "with_pluginmanager": [True, False],
+        "with_testsuite": [True, False],
+        "with_utility": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_deprecated": True,
+        "with_interconnect": True,
+        "with_main": True,
+        "with_pluginmanager": True,
+        "with_testsuite": True,
+        "with_utility": True,
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version.value) < 14:
```

Comment:
should we consider < GCC 5, < Clang 3, < Apple-Clang <I dunno> ?

Reply:
GCC 4.8.1 and clang ~3.3 are the minimal versions. For Clang I'm pretty sure no widely-used distro uses the old versions anymore (and definitely not anybody on Mac), for GCC there's just CentOS that has 4.8 (and Ubuntu 14.04, but who uses that nowadays), all other distros have newer versions.

Corrade's buildsystem will throw an error on its own if a too old version is used.

Reply:
nice explanation! thanks!